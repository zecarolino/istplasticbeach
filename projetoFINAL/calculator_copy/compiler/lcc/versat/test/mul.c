int main() {
	int a = 3, b = 3, mul, i;
	for (mul = i = 0; i < 32; i++)
		if (b & (1 << i))
			mul += a << i;
		else /* optimize if upper bits are zero */
			if (!(b >> i))
				break;
	
	return mul;
}
