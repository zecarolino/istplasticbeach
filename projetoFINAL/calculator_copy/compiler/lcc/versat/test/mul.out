VCD info: dumpfile xtop.vcd opened for output.
9
Catched Trap at time  656 clock cycles:
